File: frontend/bff/src/application/files/queries/list-files.query.ts
import type { FileKind } from '../../../cross-cutting/enums/file-kind.js';
import type { PageRequest } from '../../../cross-cutting/paging/page.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface ListFilesQuery {
  context: RequestContext;
  /** null = todos os tipos que o usuário pode ver. */
  kind: FileKind | null;
  page: PageRequest;
}

File: backend/storage-service/src/Fix.Storage.Application/UseCases/Files/Queries/ListFilesQuery.cs
using Fix.Storage.Application.Abstractions.Authorization;
using Fix.Storage.Application.Abstractions.Context;
using Fix.Storage.Application.Abstractions.Messaging;
using Fix.Storage.Application.Files.Dtos;
using Fix.Storage.Domain.AggregateRoots.Files;

namespace Fix.Storage.Application.Files.Queries;

/// <summary>Arquivos da organização, do mais recente ao mais antigo. Sem tipo: todos os que o usuário pode ver.</summary>
[RequireMembership]
public sealed record ListFilesQuery(Guid UserId, Guid OrganizationId, FileKind? Kind, int Page, int PageSize)
    : IQuery<PagedList<FileDto>>, IOrganizationRequest;
